src/main/java/com/microsoft/sqlserver/jdbc/SharedTimer.java

```diff
@@ -77,11 +81,13 @@ public synchronized void removeRef() {
      * When the caller is finished with the SharedTimer it must be released via {@link#removeRef}
      */
     public static synchronized SharedTimer getTimer() {
-        if (instance == null) {
-            // No shared object exists so create a new one
-            instance = new SharedTimer();
+        synchronized (lock) {
+            if (instance == null) {
+                // No shared object exists so create a new one
+                instance = new SharedTimer();
+            }
+            instance.refCount.getAndIncrement();
         }
-        instance.refCount += 1;
         return instance;
```

Comment:
The `return instance;` should be moved up into the `synchronized (lock) {...}` block. Otherwise there's a potential race condition if the instance is cleared by a concurrent call to `removeRef()` that completes after this synchronized block completes but prior to this method returning the value of `instance`.